File: SimTriggers/src/com/IconPippi/simtriggers/triggers/KeyTrigger.java
package com.IconPippi.simtriggers.triggers;

import java.io.IOException;

import com.IconPippi.simtriggers.EVENT;
import com.IconPippi.simtriggers.event.EventFactory;
import com.IconPippi.simtriggers.utils.Logger;

public class KeyTrigger extends Trigger {
	
	private String key;
	
	private final EventFactory eventFactory = new EventFactory();
	private final Logger logger = new Logger();
	
	protected KeyTrigger(String method) {
		super(method);
	}

	@Override
	protected void trigger() {
		if (key != null) return;
		scriptLoader.invokeFunction(method, (Object[]) null);
	}
	
	protected void trigger(String key) {
		if (key == this.key) {
			scriptLoader.invokeFunction(method, (Object[]) null);
		}
	}
	
	public void setKey(String key) {
		this.key = key;
		
		try {
			eventFactory.buildInputEvent(key, EVENT.GROUP_KEYS);
		} catch (IOException e) {
			logger.error(e.toString());
		}
	}
	
}

File: src/main/kotlin/com/simtriggers/fsx/triggers/KeyTrigger.kt
package com.simtriggers.fsx.triggers

import com.simtriggers.fsx.event.GROUP

/**
 * 05/08/2019
 * Register new key binds
 *
 * @param function Function name
 * @param keyName The key you want to bind
 *
 * @author IconPippi
 */
class KeyTrigger(private val function: String, private val keyName: String) : Trigger(function, keyName) {

    /**
     * Register the key event
     */
    init {
        eventRegister.registerClientEvent(keyName, GROUP.KEYBIND)
    }

    override fun trigger(eventName: String?) {
        if (eventName == keyName) {
            scriptLoader.invokeFunction(function, null)
        }
    }

}